### DISCUSSION TITLE
Heavy content site vs performance

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#45001] [Help]
Is there a point where you just CANNOT optimize anymore? 
I'm just wondering if you have a heavy-content site like mine https://history-maps.com/ . 
I want to provide the best content for history which means adding media, translated text, etc and this means that the data for the page, increases every time I add stuff. This questions is really basic. More content  = hit on performance. Is there a magic bullet with Nextjs or in general to optimize heavy content sites? or will I have to think of other solutions such as breaking up the content into different pages?

**Comment by icyJoseph** [comment]
A quick take would be that, media can be preloaded, https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload.

So a fetch requests, videos, images, etc, can be marked as "preload", to tell the browser to go get them ASAP, as the page loads. 

I explored that a bit here, https://github.com/vercel/next.js/discussions/42886#discussioncomment-4155003, where someone wanted to move the scripts for loading JavaScript to the end of the body, so that the defer queue would favour images and others, but through  preloading, we could remove the need to move the scripts.

This way, as you page Javascript downloads, the media needed for it is also downloaded. 

Of course compression in all payloads is important.

At the end of the day if you want to show 5 pictures of 1MB each to the users, that'll have to travel to browser, one way or another, as the HTTP TPC/IP stack allows. However, framework code, analytics, and other accessory things could be deferred, pre-loaded, etc.
